drivers/storage/portworx/portworx.go

```diff
@@ -647,6 +647,22 @@ func (p *portworx) SetDefaultsOnStorageCluster(toUpdate *corev1.StorageCluster)
 
 	setAutopilotDefaults(toUpdate)
 	setTLSDefaults(toUpdate)
+
+	if pxutil.IsFreshInstall(toUpdate) {
+		// Check for FACD variable on a new install: if so, add an annotation saying
+		// it's okay to continue.
+		facdTopologyEnv := ""
+		for _, env := range toUpdate.Spec.Env {
+			if env.Name == "FACD_TOPOLOGY_ENABLED" {
+				facdTopologyEnv = strings.ToLower(env.Value)
+				break
+			}
+		}
+
+		if facdTopologyEnv == "true" {
+			toUpdate.Annotations[pxutil.AnnotationFACDTopology] = "true"
```

Comment:
Should we make sure that annotations is not nil?
```
if toUpdate.Annotations == nil {
  toUpdate.Annotations = make(map[string]string)
}
toUpdate.Annotations[pxutil.AnnotationsFACDTopology] = "true"
```

Reply:
That gets guaranteed at the top of the function luckily, so we're good: https://github.com/libopenstorage/operator/pull/1101/files#diff-cc28f01bf9bf855945fe56dc2e65f61cf8e914889d3f93bad610daf12dff8f4eR399-R401